sdk/agentserver/azure-ai-agentserver-core/azure/ai/agentserver/core/client/tools/aio/_client.py

```diff
@@ -0,0 +1,226 @@
+		:return: The result from the tool invocation.
+		:rtype: Any
+		"""
+		payload = InvocationPayloadBuilder.build_payload(args, kwargs, {})
+		return await self._client._invoke_tool(self._descriptor, payload)
+
+	async def __call__(self, *args: Any, **kwargs: Any) -> Any:
+		return await self.invoke(*args, **kwargs)
+
+class AzureAIToolClient:
+	"""Asynchronous client for aggregating tools from Azure AI MCP and Tools APIs.
+
+	This client provides access to tools from both MCP (Model Context Protocol) servers
+	and Azure AI Tools API endpoints, enabling unified tool discovery and invocation.
+
+	:param str endpoint:
+		The fully qualified endpoint for the Azure AI Agents service.
+		Example: "https://<resource-name>.api.azureml.ms"
+	:param credential:
+		Credential for authenticating requests to the service.
+		Use credentials from azure-identity like DefaultAzureCredential.
+	:type credential: ~azure.core.credentials.TokenCredential
+	:keyword str agent_name:
+		Name of the agent to use for tool operations. Default is "$default".
+	:keyword List[Mapping[str, Any]] tools:
+		List of tool configurations defining which tools to include.
+	:keyword Mapping[str, Any] user:
+		User information for tool invocations (object_id, tenant_id).
+	:keyword str api_version:
+		API version to use when communicating with the service.
+		Default is the latest supported version.
+	:keyword transport:
+		Custom transport implementation. Default is RequestsTransport.
+	:paramtype transport: ~azure.core.pipeline.transport.HttpTransport
+
+	"""
+
+	def __init__(
+		self,
+		endpoint: str,
+		credential: "AsyncTokenCredential",
+		**kwargs: Any,
+	) -> None:
+		"""Initialize the asynchronous Azure AI Tool Client.
+
+		:param str endpoint: The service endpoint URL.
+		:param credential: Credentials for authenticating requests.
+		:type credential: ~azure.core.credentials.TokenCredential
+		:keyword kwargs: Additional keyword arguments for client configuration.
+		"""
+		self._config = AzureAIToolClientConfiguration(
+				endpoint,
+				credential,
+				**kwargs,
+			)
+		
+		_policies = kwargs.pop("policies", None)
+		if _policies is None:
+			_policies = [
+				policies.RequestIdPolicy(**kwargs),
+				self._config.headers_policy,
+				self._config.user_agent_policy,
+				self._config.proxy_policy,
+				policies.ContentDecodePolicy(**kwargs),
+				self._config.redirect_policy,
+				self._config.retry_policy,
+				self._config.authentication_policy,
+				self._config.custom_hook_policy,
+				self._config.logging_policy,
+				policies.DistributedTracingPolicy(**kwargs),
+				policies.SensitiveHeaderCleanupPolicy(**kwargs) if self._config.redirect_policy else None,
+				self._config.http_logging_policy,
+			]
+		self._client: AsyncPipelineClient = AsyncPipelineClient(base_url=endpoint, policies=_policies, **kwargs)
+	
+		# Initialize specialized clients with client and config
+		self._mcp_tools = MCPToolsOperations(client=self._client, config=self._config)
```

Comment:
if these are clients do we want them under /operations? Can we keep their operations under operations and pull out the client to the same level as _client.py